projects/igniteui-angular/src/lib/grids/common/crud.service.ts

```diff
@@ -251,6 +251,11 @@ export class IgxCellCrudState {
         if (!this.cell) {
             return;
         }
+        // this is needed when we are not using ngModel to update the editValue
+        // so that the change event of the inlineEditorTemplate is hit before
+        // trying to update any cell
+        const activeElement = document.activeElement as HTMLElement;
```

Comment:
Accessing the document directly might cause issues with shadow dom. Might be best to get the root node instead like in the forof: https://github.com/IgniteUI/igniteui-angular/blob/master/projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts#L1054

Reply:
Haha, posted on time.
True actually, I missed the part where the grid itself is under a shadow root and document will be wrong too. So we need a node ref (from the cell perhaps) we can get the root from.